### ISSUE TITLE
[dev] Selecting programmatically but having an issue dealing with exponents

### ISSUE THREAD

**Issue by Darkneon** [mathquill/mathquill#437]
I have the following scenario. I have an expression with an exponent, let's say 8^2, when I click between the 8 and the 2 I want to select both numbers.

My approach is to find the node of 8 and the node of 2 and then call ```selectChildren(left, right)``` 

![screenshot from 2015-07-10 17 00 04](https://cloud.githubusercontent.com/assets/654727/8628971/26f73fe2-2725-11e5-9fa8-897860827a1b.png)

but I get the error
```
Uncaught Error: prayer failed: withDir and oppDir have the same parent
```

So my question is, what's the proper approach for selecting when the parents are not the same?

**Comment by laughinghan** [comment]
Right, those are both descendants but not both children. The way selection works when you drag from the 8 to the 2 is that `Cursor::startSelection()` is called when you mousedown, and then `Cursor::select()` will always select from there to where the cursor has moved to.

There used to be a `Fragment.between()` that would find the lowest common ancestor (LCA), but I got rid of it for a simpler algorithm that I inlined into `Cursor::select()`: 3dc7dcea199653a7a8aac18f9b3bb8e5bb1723d4 of #266. If you want a fragment between 8 and 2 that's not a selection, you can run the algorithm yourself, it's not that complicated: collect the ids of ancestors of the 8, then go through the ancestors of the 2 until you find one whose parent is an ancestor of 8, now you have the relevant siblings necessary to create a fragment.
